library/std/src/thread/mod.rs

```diff
@@ -1759,6 +1759,19 @@ impl<T> JoinHandle<T> {
     /// }).unwrap();
     /// join_handle.join().expect("Couldn't join on the associated thread");
     /// ```
+    ///
+    /// # Notes
+    ///
+    /// This function has the same minimal guarantee regarding "foreign" unwinding operations (e.g.
```

Comment:
Is `join` the right place to document this? After all, it's `spawn` that sets up the catch_unwind, and indeed an abort can happen due to that even if one never calls `join`. So `spawn` seems like the better place?

Reply:
No, you're right, the recovery happens regardless of whether the thread is joined or detached. So `join` is only relevant as a way to check whether the thread panicked. I'll fix this.

Reply:
Fixed.